src/variety/railpool.js

```diff
@@ -0,0 +1,659 @@
+					cm += pstr;
+					count = 1;
+				} else if (pstr !== "" || count === 16) {
+					cm += (19 + count).toString(36) + pstr;
+					count = pstr === "" ? 0 : 1;
+				}
+			}
+			if (count > 1) {
+				cm += (19 + count).toString(36);
+			}
+
+			this.outbstr += cm;
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeQnums_railpool();
+			this.decodeEmpty();
+			this.decodeAreaRoom();
+			this.decodeBorderLine();
+		},
+		encodeData: function() {
+			this.encodeQnums_railpool();
+			this.encodeEmpty();
+			this.encodeAreaRoom();
+			this.encodeBorderLine();
+		},
+		// from tapaloop.js
+		decodeQnums_railpool: function() {
+			this.decodeCell(function(cell, ca) {
+				if (ca !== ".") {
+					cell.qnums = [];
+					var array = ca.split(/,/);
+					for (var i = 0; i < array.length; i++) {
+						cell.qnums.push(array[i] !== "-" ? +array[i] : -2);
+					}
+				}
+			});
+		},
+		encodeQnums_railpool: function() {
+			this.encodeCell(function(cell) {
+				if (cell.qnums.length > 0) {
+					var array = [];
+					for (var i = 0; i < cell.qnums.length; i++) {
+						array.push(cell.qnums[i] >= 0 ? "" + cell.qnums[i] : "-");
+					}
+					return array.join(",") + " ";
+				} else {
+					return ". ";
+				}
+			});
+		},
+		// from country.js
+		decodeEmpty: function() {
+			this.decodeCell(function(cell, ca) {
+				if (ca === "*") {
+					cell.ques = 7;
+				}
+			});
+		},
+		encodeEmpty: function() {
+			this.encodeCell(function(cell) {
+				if (cell.ques === 7) {
+					return "* ";
+				} else {
+					return ". ";
+				}
+			});
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkBranchLine",
+			"checkCrossLine",
+
+			"checkUncluedLength",
+			"checkMissingLength",
+			"checkIncompatibleSets",
+
+			"checkDeadendLine+",
+			"checkOneLoop",
+			"checkNoLine"
+		],
+
+		checkUncluedLength: function() {
+			var result = true,
+				rooms = this.board.roommgr.components;
+			for (var r = 0; r < rooms.length; r++) {
+				var clist = rooms[r].clist,
+					clueSet = rooms[r].clist.getClueSet();
+				if (clueSet.length === 0) {
+					continue;
+				}
+				if (clueSet.includes(-2)) {
+					// question marks are checked in checkIncompatibleSets
+					continue;
+				}
+				for (var i = 0; i < clist.length; i++) {
+					var cell = clist[i];
+
+					var horiz = cell.getSegment(true);
+					var vert = cell.getSegment(false);
+
+					if (horiz.length > 0 && !clueSet.includes(horiz.length)) {
+						result = false;
+						if (this.checkOnly) {
+							break;
+						}
+						horiz.seterr(1);
+					}
+
+					if (vert.length > 0 && !clueSet.includes(vert.length)) {
+						result = false;
+						if (this.checkOnly) {
+							break;
+						}
+						vert.seterr(1);
+					}
+				}
+
+				if (this.checkOnly && !result) {
+					break;
+				}
+			}
+			if (!result) {
+				this.failcode.add("segUnclued");
```

Comment:
If you call `setnoerr()` on every border on the board, it grays out all non-errored lines.